geo/src/array.rs

```diff
@@ -0,0 +1,29 @@
+//! `ExtensionArray` wrapper for arrays that hold geospatial data types.
+
+use vortex::arrays::ExtensionArray;
+use vortex::error::{VortexError, VortexResult};
+use vortex::variants::ExtensionArrayTrait;
+
+use crate::{GeoMetadata, GeometryType};
+
+/// Holder for what is known to be one of the blessed extension array types.
+pub enum GeometryArray<'a> {
```

Comment:
Is this sort of equivalent to Canonical? I wonder if there's value in allowing an ExtensionDType to canonicalize into a custom impl of some `dyn ExtensionArray`?

It's a bit more complex than Arrow extension types, because it means dtypes must come with a vtable, instead of just some metadata bytes. Not sure yet.